telos-frontend/src/components/habit-tracker/DashboardHabitTracker.jsx

```diff
@@ -1,32 +1,325 @@
+import { useState } from 'react';
 import AddIcon from '@material-ui/icons/Add';
-import IconButton from '@material-ui/core/IconButton';
+import {
+  IconButton,
+  Box,
+  Button,
+  TextField,
+  InputLabel,
+  MenuItem,
+  Select,
+  Radio,
+  RadioGroup,
+  FormControlLabel,
+} from '@material-ui/core';
 import './DashboardHabitTracker.css';
 
-const DashboardHabitTracker = () => (
-  <div className="container">
-    <div className="HabitTitle">
-      <h2>hello world</h2>
-      <span className="AddIcon">
-        <IconButton>
-          <AddIcon />
-        </IconButton>
-      </span>
-    </div>
-    <div className="HabitContent">
-      <p>Habit description, Do Task X, Y many times per week</p>
-    </div>
-    <div className="WeekSchedule">
-      <ul>
-        <li>M</li>
-        <li>T</li>
-        <li>W</li>
-        <li>T</li>
-        <li>F</li>
-        <li>S</li>
-        <li>S</li>
-      </ul>
+const DashboardHabitTracker = () => {
+  const [habitTitle, setTitle] = useState('Habit1');
+  const [tempTitle, setTempTitle] = useState(habitTitle);
+  const [habitDes, setDes] = useState('Habit description, Do Task X, Y many times per week');
+  const [tempDes, setTempDes] = useState(habitDes);
+
+  const [monStatus, setMonStatus] = useState(false);
+  const [tueStatus, setTueStatus] = useState(false);
+  const [wedStatus, setWedStatus] = useState(false);
+  const [thuStatus, setThuStatus] = useState(false);
+  const [friStatus, setFriStatus] = useState(false);
+  const [satStatus, setSatStatus] = useState(false);
+  const [sunStatus, setSunStatus] = useState(false);
+
+  const [weeks, setWeeks] = useState('1');
+  const [startDate, setStartDate] = useState('2021-01-01');
+  const [endType, setEndType] = useState('Never');
+  const [endDate, setEndDate] = useState('');
+
+  const [isEdit, setEditStatus] = useState(false);
+
+  const handleChangeOnName = (event) => {
+    setTempTitle(event.target.value);
+  };
+
+  const handleChangeOnDes = (event) => {
+    setTempDes(event.target.value);
+  };
+
+  const handleChangeOnWeeks = (event) => {
+    setWeeks(event.target.value);
+  };
+
+  const handleChangeOnStartDate = (event) => {
+    setStartDate(event.target.value);
+  };
+
+  const handleChangeOnEndDate = (event) => {
+    setEndDate(event.target.value);
+  };
+
+  const handleChangeOnEndType = (event) => {
+    setEndType(event.target.value);
+  };
+
+  const saveChange = () => {
+    setTitle(tempTitle);
+    setDes(tempDes);
+  };
+
+  return (
+    <div>
+      <div className="container">
+        <div className="HabitTitle">
+          <h2>{habitTitle}</h2>
+          <span className="AddIcon">
+            <IconButton onClick={() => setEditStatus(true)}>
+              <AddIcon />
+            </IconButton>
+          </span>
+        </div>
+        <div className="HabitContent">
+          <p>{habitDes}</p>
+        </div>
+        <div className="WeekSchedule">
```

Comment:
Could this be separated into a separate subcomponent? You could pass the states and mutators through and it could be used further down to reduce duplication

Reply:
Do you mean to create another component.jsx file and import it to this file?

Reply:
> Could this be separated into a separate subcomponent? You could pass the states and mutators through and it could be used further down to reduce duplication

Do you mean to create another component.jsx file and import it to this file?

Reply:
Yeah you can create new components and store them in habit-tracker/